### ISSUE TITLE
Not possible to autocreate documentation for multiple ArgumentNullExceptions

### ISSUE THREAD

**Issue by jesperborasmussen** [CSharpAnalyzers/ExceptionalReSharper#33]
I multple parameters throws ArgumentNullException, it is only possible to autogenerate documentation for one of the. The highlighting of the missing documentation disappears, after adding it for the first ArgumentNullException. The Following code...

`public Guid Import(Stream fileData, string fileName)
{
            if (fileData == null) throw new ArgumentNullException("fileData");
            if (fileName == null) throw new ArgumentNullException("fileName");
}`

Generates the following documentation:

`/// <exception cref="ArgumentNullException"><paramref name="fileData"/> is <see langword="null" />.</exception>
public Guid Import(Stream fileData, string fileName, Guid? cachedFileId)
{
    if (fileData == null) throw new ArgumentNullException("fileData");
    if (fileName == null) throw new ArgumentNullException("fileName");
}`

**Comment by pedromagueija** [comment]
Hi Jasper,

The exception being documented is the main point, not the arguments being checked. Regardless of which parameter is null, the exception thrown is the same, thus the check passes (the exception is documented).

In this particular case, the message could be manually edited to reflect both parameters that are checked.
Do you agree?

**Comment by jesperborasmussen** [comment]
Hi Pedro

I agree that the exception could be manually edited. But that partly defeat the purpose of autogenerating documentation. I still think that the documentation must in this case contain which parameters that can throw an ArgumentNullException, because this also shows which doesn't.

**Comment by pedromagueija** [comment]
I understand your point.

So something like this would be ok for you?

```
/// <exception cref="ArgumentNullException">
///   <paramref name="fileData"/> is <see langword="null" /> or
///   <paramref name="fileName"/> is <see langword="null" />.
/// </exception>
public Guid Import(Stream fileData, string fileName, Guid? cachedFileId)
{
    if (fileData == null) throw new ArgumentNullException("fileData");
    if (fileName == null) throw new ArgumentNullException("fileName");
}


```

**Comment by jesperborasmussen** [comment]
Yes that is how i would write it except that Microsoft writes it like this "-or-" in their documentation.
